src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -152,7 +154,8 @@ public function onKernelResponse(ResponseEvent $event)
             $request = $event->getRequest();
             $requestSessionCookieId = $request->cookies->get($sessionName);
 
-            if ($requestSessionCookieId && $session->isEmpty()) {
+            $isSessionEmpty = $session->isEmpty() && empty($_SESSION); // checking native $_SESSION to keep compatibility to php bridge sessions
```

Comment:
check on global véritable`$_SESSION` sounds dubious..
why this check is not pretty of the method `isEmpty()`?

Reply:
Not sure if thats a good idea. As $_SESSION should only be kept in mind here. As $_SESSION is only correctly updated after the `save` method is called (some lines above). If we move it to HttpFoundation component to handle keep $_SESSION in mind for isEmpty the check will not end in a correct result for not yet saved bags.

Reply:
Would be a good idea to have a method like `isDestroyaable(): bool` on each session handler implementation? In that case, each handler could call his own `isEmpty()` implementation, and in the case of the php bridge it could call `isEmpty()` and `empty($_SESSION)`

Reply:
The difficulty is that would extend the public API, which Symfony is very strict about it and doing only in minor releases. When we would do that we could also directly store the $destroy state in the Session instance of the HttpFoundation component. And then go with Session::isDestroyed(): bool.
Personally would like that lifetime and other things also live on the Session variable and not on some options, but that's another topic.
Still our other option would be skip the whole things when the PHPBridge Session is active and so go with the behaviour of Symfony 5.3 where php will write and remove the session headers. Disadvantage is that the php bridge session not work then in CLI context (session based tests, runtimes like swoole roadrunner not working). To skip that we would need have a boolean flag injected which is set when php session bridge is active to another value.

Reply:
`isDestroyable` is a good idea, but cannot be used for a bug fix (this PR)

I don't like checking `$_SESSION` (It sounds weird to check the global variable in case somebody changed it), but I don't know how to fix this case without it. 

ping @symfony/mergers

Reply:
Works for me as-is, `$_SESSION` is already used in `reset()` l174.